esp-client/src/main.cpp

```diff
@@ -7,46 +7,85 @@
 
 #define WIPE false // if true the nvs partition and all saved values will be wiped
 
+// ----------------------- MQTT-Server Settings ---------------------  //
 #define SSID "Postmelder-Wifi"
 IPAddress mqttServer(10, 42, 0, 1);
 #define MQTT_USER "MQTTBroker"
 #define MQTT_PASS "postmelder"
 
+// -----------------------  SCALE Settings --------------------------  //
 #define SCALE_THRESHOLD 2.0 // value in grams, above or below which no change will be reported in grams
 #define SCALE_DATA_PIN 32
 #define SCALE_CLOCK_PIN 33
 
+// ------------------------------- Pinout ----------------------------  //
+#define R_LED_PIN 0
+#define G_LED_PIN 2
+#define B_LED_PIN 15
+
+// ------------------------- PMW Blinking ------------------------------ //
+#define CHANNEL_BLAU 0
+#define CHANNEL_ROT 1
+#define BLINKEN_EIN 100
+#define BLINKEN_AUS 0
+
 WiFiClient wiFiClient;
 PubSubClient client(wiFiClient);
-
 HX711 scale;			 // scale object
 Preferences preferences; // preferences object
-
-float weight = 0; // scale measurement variables
-
-float scaleValue;	   // calibrated scale values -> Flash
-long scaleOffset;	   // -> Flash
-bool scaleInitialised; // saves whether the scale has already been calibrated-> Flash
-
+float weight = 0;		 // scale measurement variables
+float scaleValue;		 // calibrated scale values -> Flash
+long scaleOffset;		 // -> Flash
+bool scaleInitialised;	 // saves whether the scale has already been calibrated-> Flash
 const String MAC = WiFi.macAddress();
 bool connectedWithNode = false;
 bool isServerOnline = false;
 
+// Function declarations
+void updateLEDs();
+void setStateOccupied(bool value);
+void setStateErrorCommunication(bool value);
+void setStateErrorScale(bool value);
+void setStateInit(bool value);
 void callback(char *topic, byte *message, unsigned int length);
 void reconnect();
 void sendWeight(float weight);
 void calibrateScale();
 float readScale();
 
+/*
+	Bits: 		Func
+	0	:		INIT
+	1	:		OCCUPIED
+	2-5	:		reserver
+	6	:		ERR_Scale
+	7	: 		ERR_Communication
+*/
+char state = 0b000;
+
 void setup()
 {
 #if WIPE			   // if wipe is defined
 	nvs_flash_erase(); // format nvs-partition
 	nvs_flash_init();  // initialise nvs-partition
 #endif
 
+	setStateOccupied(false);
+	setStateErrorCommunication(false);
+	setStateInit(true);
+
```

Comment:
Könnte man eventuell in `setStateInit` automatisch alle bits so setzen, dann wäre es Idiotensicherer